shared-articles/what does dcdd do.md

```diff
@@ -0,0 +1,75 @@
+
+Ultimately, blockchain is a network, pretty much like any other network. Where one computer is connected to another, and then in turn, each computer added is connected to all others (or as many as possible). 
+
+The immutable nature of the information being transmitted and stored is the distinct difference between blockchain and a conventional network. Effectively, this is a book of accounts, better known as a ledger, which is synchronised across all fully participating computers (nodes). To participate, each computer needs to carry the ledger in its entirety and synchronise with other computers and the pool of new transactions. 
+
+The ledger is made up of individual pages which are known as blocks, and each block is linked with a unique code which makes it hard to reverse or manipulate the historic information stored. 
+
+The software that makes this possible on the Decred blockchain is dcrd, which is known as a full node implementation of Decred (written in Go – golang). It acts as a fully validating chain daemon for the Decred cryptocurrency. A daemon is a computer program that runs as a background process, rather than being under the direct control of an interactive user. You can consider dcrd to be your personal Decred blockchain server.
+
+## dcrd responsibilities:
+dcrd is the backbone of Decred’s peer to peer network and has specific processes to conduct to keep the network functioning. Running a full node with dcrd contributes to the overall security of the network and increases the available paths for transactions and blocks to relay.
+
+* dcrd maintains the entire past transactional ledger of Decred
+* Allows relaying of transactions to other Decred nodes around the world
+* Maintains a transaction pool
+* Relays newly mined blocks
+* and finally ensures all transactions follow the rules
+
+In terms of benefits, since dcrd fully validates every block and transaction, it provides the highest security and privacy possible when used with a wallet that’s also in full validation mode. This is ideal for individuals, businesses, and services that need the most reliable and accurate data about transactions.
+
+## Node Hierarchy
+As Decred builds out its infrastructure, it becomes apparent that the underlying security is a high priority. Each time a new service is added, it increases the opportunity for node participation. This is done on multiple levels, and it’s important to note that not every node is equal.
+
+**A top tier fully validating node** is a dcrd unit run either as a stand-alone piece of software or as part of a service for 24 hours, 7 days a week, 365 days a year. This node has an inbound port open, so it can be publicly accessible for other nodes to connect to. This full node is the one that adds security to the chain and makes the project more decentralised. 
```

Comment:
I feel like these are the most important dcrd distinctions:

* Uptime
    * High-uptime (running 24/7/365, or close to it)
    * Low-uptime (running intermittently)
* With inbound connections
    * No (default)
    * Yes (requires the user to open up a port in their firewall)

Theoretically you could also add the distinction that some dcrd nodes are also miners, while most nodes are not mining, but I honestly don't think that's helpful since most people reading this won't be mining. And also, most miners are probably running their own bespoke software that's customized to their specific hardware.

But anyway, it doesn't really seem helpful to me to talk about a hierarchy between these types of nodes:
* High-uptime with inbound connections
* High-uptime without inbound connections
* Low-uptime with inbound connections
* Low-uptime without inbound connections

Each of those nodes are full nodes. They're all doing the work of a full node. It's just that, of the list of those 4 options, the one that supports the network the best is the `High-uptime with inbound connections` full-node. And as @jholdstock said, if you're staking, you're incentivized to BOTH have high-uptime (so you won't miss a vote) AND allow inbound connections (so you have more peers, meaning you receive messages propagating through the network earlier, meaning you won't miss a vote).

Staking literally incentivizes high-uptime full nodes with inbound connections. It's awesome.

But yeah, anyway, all 4 of those types of nodes are all full nodes.